util/topsql/reporter/single_target.go

```diff
@@ -53,7 +54,7 @@ func NewSingleTargetDataSink(registerer DataSinkRegisterer) *SingleTargetDataSin
 		conn:       nil,
 		sendTaskCh: make(chan sendTask, 1),
 
-		registered: false,
+		registered: atomic.NewBool(false),
```

Comment:
why do we need this change?

Reply:
`Close` accessing this variable leads to concurrent accessing.